cmd2.py

```diff
@@ -1480,6 +1481,28 @@ def complete_help(self, text, line, begidx, endidx):
         completions.sort()
         return completions
 
+    # noinspection PyUnusedLocal
+    def sigint_handler(self, signum, frame):
+        """Signal handler for SIGINTs which typically come from Ctrl-C events.
+
+        If you need custom SIGINT behavior, then override this function.
+
+        :param signum: int - signal number
+        :param frame
+        """
+        # Save copy of pipe_proc since it could theoretically change while this is running
+        pipe_proc = self.pipe_proc
+        if pipe_proc is not None:
+            pipe_proc.terminate()
+
+        # Re-raise a KeyboardInterrupt so other parts of the code can catch it
+        raise KeyboardInterrupt("Got a keyboard interrupt within a Python script")
```

Comment:
Is this supposed to check the self._in_py flag before raising the exception?

Reply:
Initially I made this method much more complicated where it had a couple checks, one for the flag you mention and onother for the case where we were blocking waiting for input from the user.  But then I thought about it some more and I think it is safe to always re-raise the KeyboardInterrupt.

I manually treated on both macOS and Windows and was happy with the behavior in all cases that I thought of testing.